### PR TITLE
Add some validation checks

### PR DESCRIPTION
**Pull request by nalind**
Add checks using gofmt, gometalinter, and git-validate, and fixup the things that they flag.

### CODE REVIEW

File: import.go [hunk 1]
```diff
@@ -26,13 +25,13 @@ func importBuilder(store storage.Store, options ImportOptions) (*Builder, error)
...
 		}
-		src, err := ref.NewImage(systemContext)
-		if err != nil {
-			return nil, fmt.Errorf("error instantiating image: %v", err)
+		src, err3 := ref.NewImage(systemContext)
+		if err3 != nil {
+			return nil, fmt.Errorf("error instantiating image: %v", err3)
 		}
 		defer src.Close()
 		config, err = src.ConfigBlob()
```
Comment (TomSweeneyRedHat) [inline import.go]: Not an issue, but I'm not getting why you added numbers to some but not all 'err' variables.

File: cmd/buildah/config.go [hunk 2]
```diff
@@ -129,8 +129,8 @@ func updateConfig(builder *buildah.Builder, c *cli.Context) {
 		}
 	}
 	if c.IsSet("volume") {
-		for _, volSpec := range c.StringSlice("volume") {
-			builder.Volumes = append(builder.Volumes, volSpec)
+		if volSpec := c.StringSlice("volume"); len(volSpec) > 0 {
+			builder.Volumes = append(builder.Volumes, volSpec...)
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/config.go]: Neat trick using the ..., will have to try and remember that.

File: import.go (hunk 1 above)
Comment (nalind) [inline import.go]: These avoid shadowing warnings from one of the linters, by removing cases where we declare a variable in a block when a variable with that same name already exists in an enclosing block.  https://golang.org/ref/spec#Short_variable_declarations also covers why we don't have to do this when we redeclare a variable from the same block.

File: cmd/buildah/config.go (hunk 2 above)
Comment (nalind) [inline cmd/buildah/config.go]: Agreed, it's a very handy language feature.

**Comment by nalind** [comment]
[(8) [Buildah] Add "dockah build-using-dockerfile" aka "buildah bud"](https://trello.com/c/AyBXqoRY/443-8-buildah-add-dockah-build-using-dockerfile-aka-buildah-bud)

File: import.go (hunk 1 above)
Comment (TomSweeneyRedHat) [inline import.go]: Ty for the 411!

**Comment by nalind** [comment]
Rebased.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ c842847

**Comment by nalind** [comment]
@rh-atomic-bot retry

**Comment by nalind** [comment]
Reworked some of the install targets to merge install-tools and .install.md2man and avoid having "make all" install things outside of the build tree.

**Comment by nalind** [comment]
@TomSweeneyRedHat, @rhatdan, PTAL again for that last patch, which is kind of small to be on its own, since we're already messing around with the top-level Makefile.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ 540eb22

**Pull request closed without merging**